crates/bevy_ecs/src/schedule/state.rs

```diff
@@ -185,56 +233,1212 @@ impl<S: States> NextState<S> {
+    ) {
+        {
+            let system = |world: &mut World| {
+                let state_set = world.get_resource::<State<S::RawState>>();
+                let new_state = if let Some(state_set) = S::convert_to_usable_state(state_set) {
+                    T::compute(state_set)
+                } else {
+                    None
+                };
+                internal_apply_state_transition(world, new_state);
+            };
+            let label = ComputeComputedState::<T>::default();
+            schedules.insert({
+                let mut schedule = Schedule::new(label);
+                schedule.add_systems(system);
+                schedule
+            });
+        }
+
+        {
+            let system = |mut transitions: ResMut<StateTransitionSchedules>| {
+                transitions
+                    .dependant_schedules
+                    .entry(<T as InnerStateSet>::DEPENDENCY_DEPTH)
+                    .or_default()
+                    .insert(ComputeComputedState::<T>::default().intern());
+            };
+            let label = ComputeDependantStates::<S::RawState>::default();
+            match schedules.get_mut(label.clone()) {
+                Some(schedule) => {
+                    schedule.add_systems(system);
+                }
+                None => {
+                    let mut schedule = Schedule::new(label);
+                    schedule.add_systems(system);
+                    schedules.insert(schedule);
+                }
+            }
+        }
+    }
+
+    fn register_state_exist_systems_in_schedule<T: SubStates<SourceStates = Self>>(
+        schedules: &mut Schedules,
+    ) {
+        {
+            let system = |world: &mut World| {
+                let state_set = world.get_resource::<State<S::RawState>>();
+                let new_state = if let Some(state_set) = S::convert_to_usable_state(state_set) {
+                    T::exists(state_set)
+                } else {
+                    None
+                };
+                match new_state {
+                    Some(value) => {
+                        if !world.contains_resource::<State<T>>() {
+                            internal_apply_state_transition(world, Some(value));
+                        }
+                    }
+                    None => internal_apply_state_transition::<T>(world, None),
+                };
+            };
+            let label = ComputeComputedState::<T>::default();
+            schedules.insert({
+                let mut schedule = Schedule::new(label);
+                schedule.add_systems(system);
+                schedule
+            });
+        }
+
+        {
+            let system = |mut transitions: ResMut<StateTransitionSchedules>| {
+                transitions
+                    .dependant_schedules
+                    .entry(T::DEPENDENCY_DEPTH)
+                    .or_default()
+                    .insert(ComputeComputedState::<T>::default().intern());
+            };
```

Comment:
Is there a way to reduce the cost on this? Running a full schedule (including their own executors) may be 100% overkill of most states. Could we reduce this to a list of either systems that are just run in sequence? A fast way to ensure that we don't add a new multithreaded exeuctor each of these is just to have each of the ComputeDependantStates schedules use a Simple executor, since all of these systems don't really need handle deferred commands or run in parallel.

Reply:
Still working on a solution to this - I have an idea that I'm trying, but have a lot on my plate for now, so it might take alittle longer

Reply:
LMK if the new solution works!